platform/eureka/controller.go

```diff
@@ -26,34 +26,38 @@ import (
 type serviceHandler func(*model.Service, model.Event)
 type instanceHandler func(*model.ServiceInstance, model.Event)
 
-type controller struct {
+// Controller represents state of Eureka registry
+type Controller struct {
 	interval         time.Duration
 	serviceHandlers  []serviceHandler
 	instanceHandlers []instanceHandler
 	client           Client
 }
 
 // NewController instantiates a new Eureka controller
-func NewController(client Client, interval time.Duration) model.Controller {
-	return &controller{
+func NewController(client Client, interval time.Duration) *Controller {
+	return &Controller{
 		interval:         interval,
 		serviceHandlers:  make([]serviceHandler, 0),
 		instanceHandlers: make([]instanceHandler, 0),
 		client:           client,
 	}
 }
 
-func (c *controller) AppendServiceHandler(f func(*model.Service, model.Event)) error {
+// AppendServiceHandler implements controller operation
+func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) error {
 	c.serviceHandlers = append(c.serviceHandlers, f)
 	return nil
 }
 
-func (c *controller) AppendInstanceHandler(f func(*model.ServiceInstance, model.Event)) error {
+// AppendInstanceHandler implements controller operation
```

Comment:
you wouldn't need these comments if controller was returned as `model.Controller`